File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-events/src/services/eventsService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { EventsPageResult } from "../types/eventsModels.js";

/**
 * Events service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface EventsService {
  readonly capability: "events";
}

export function createEventsService(client: AppstoreAppClient): EventsService {
  void client;
  return { capability: "events" };
}

export function emptyEventsResult(): EventsPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_events/lib/src/services/events_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/events_models.dart';

/// Events service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class EventsService {
  const EventsService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'events';

  EventsPageResult<Never> empty() =>
      const EventsPageResult<Never>(items: <Never>[]);
}
